### DISCUSSION TITLE
Which model is used when using the 'merge responses' function?

### DISCUSSION THREAD

**Discussion by ArnaudPolitico** [open-webui/open-webui#6640] [Q&A] [answered]
Everything is in the title; I haven't found anything in the documentation that mentions this.
It would also be interesting to be able to choose which model performs this function, as some AI models are indeed better suited for this type of task than others.

There might be part of the answer here, but it’s not certain. (https://github.com/open-webui/open-webui/discussions/4934)
It’s suggested that the model that merges the responses is the last one in the list. However, is that really the case, or is it actually the last model that responded that merges the answers? Ollama doesn’t seem to strictly follow the order of models and sometimes selects them at random.  

Thanks a lot.

**Comment by ArnaudPolitico** [comment]
I’m partly answering my own question. Contrary to what’s mentioned in this discussion, it’s not the last AI model in the list that’s used, but rather the last one that responded.

As shown in my screenshot, the last model in the list is Llama 3.2, but judging by resource usage, it appears to be one of the other four models instead.
![2024-11-02Openwebui](https://github.com/user-attachments/assets/a3320879-b95d-4e9c-96b8-fc4d8ba5cffc)

It’s unfortunate, as I would really like to be able to choose which model is used to merge multiple responses, and also to modify the prompt given for merging. 

Currently, the prompt is as follows (according to [line 1836](https://github.com/open-webui/open-webui/blob/7228b39064ac28e1240bf8998f2a35535c6f7ef5/backend/open_webui/main.py#L1836C1-L1836C478) ):
`Your task is to synthesize these responses into a single, high-quality response. It is crucial to critically evaluate the information provided in these responses, recognizing that some of it may be biased or incorrect. Your response should not simply replicate the given answers but should offer a refined, accurate, and comprehensive reply to the instruction. Ensure your response is well-structured, coherent, and adheres to the highest standards of accuracy and reliability.`

Depending on the situation, this isn’t always appropriate.

**Comment by ArnaudPolitico** [comment, marked as answer]
@tjbck just answered on another thread ( https://github.com/open-webui/open-webui/discussions/6538#discussioncomment-11149986 ).
Here there are the ArenaMod assiciated documentation : https://docs.openwebui.com/tutorials/evaluation/

**Comment by GrayXu** [comment]
I deployed using an API, and the model used for merging is the task model, which is also used to generate titles and tags.   

I also think it would be much better if we could modify the merge prompt.

**Comment by homjay** [comment]
I found the merger result to be randomly generated. Suppose you ask two models to randomly generate two animal names. Then, when asked to merge, the results might yield one animal name, or sometimes two, but the outcome is random.

Perhaps I misunderstand the merge function? Custom prompt support is important.
